src/SSHDebugPS/AD7/AD7Port.cs

```diff
@@ -1,26 +1,24 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.VisualStudio.Debugger.Interop;
-using Microsoft.VisualStudio.Debugger.Interop.UnixPortSupplier;
-using Microsoft.VisualStudio.OLE.Interop;
 using System;
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
-using System.Text;
 using System.Threading;
-using System.Threading.Tasks;
+using Microsoft.VisualStudio.Debugger.Interop;
+using Microsoft.VisualStudio.Debugger.Interop.UnixPortSupplier;
+using Microsoft.VisualStudio.OLE.Interop;
 
 namespace Microsoft.SSHDebugPS
 {
-    internal class AD7Port : IDebugPort2, IDebugUnixShellPort, IDebugPortCleanup, IDebugGdbServerAttach, IConnectionPointContainer, IConnectionPoint
+    internal abstract class AD7Port : IDebugPort2, IDebugUnixShellPort, IDebugPortCleanup, IConnectionPointContainer, IConnectionPoint
     {
         private readonly object _lock = new object();
         private readonly AD7PortSupplier _portSupplier;
-        private string _name;
+        protected string _name;
         private readonly Lazy<Guid> _id = new Lazy<Guid>(() => Guid.NewGuid(), LazyThreadSafetyMode.ExecutionAndPublication);
-        private Connection _connection;
+        protected Connection _connection;
```

Comment:
>protected Connection _connection; [](start = 8, length = 33)

Having this as a protected field is kind of weird - how are derived classes supposed to know to set it in their implementation of GetConnection? Unless there's a good reason not to, I'd have the base class handle the caching.